### PR TITLE
Add back the governance module

### PR DESCRIPTION
**Pull request by liamsi**
## Description

Basically copy & paste everything gov related from a fresh starport app.

related to: https://github.com/celestiaorg/celestia-app/issues/168

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: app/app.go [hunk 1]
```diff
@@ -297,6 +310,15 @@ func New(
 		appCodec, keys[ibchost.StoreKey], app.GetSubspace(ibchost.ModuleName), app.StakingKeeper, app.UpgradeKeeper, scopedIBCKeeper,
 	)
 
+	// register the proposal types
+	govRouter := govtypes.NewRouter()
+	govRouter.AddRoute(govtypes.RouterKey, govtypes.ProposalHandler).
+		AddRoute(paramproposal.RouterKey, params.NewParamChangeProposalHandler(app.ParamsKeeper)).
+		// TODO: do we want a community pool at all? If not, does it even make sense to include that NewCommunityPoolSpendProposalHandler?
```
Comment (liamsi) [inline app/app.go:317]: Let's discuss the community pool in a separate issue/github dicsussion thread. I'll remove the todo after we have that issue/discusssion.

**Review by evan-forbes — APPROVED** [review]
Yay! Thanks for adding this back in!

**Review by evan-forbes — CHANGES_REQUESTED** [review]
whoops, sorry, reviewed my phone and got too excited. setting a block so we don't accidentally merge before we set the gov module account.

**Comment by liamsi** [comment]
> before we set the gov module account

What do you mean?

File: app/app.go [hunk 2]
```diff
@@ -139,6 +139,7 @@ var (
 	maccPerms = map[string][]string{
 		authtypes.FeeCollectorName:     nil,
 		distrtypes.ModuleName:          nil,
+		govtypes.ModuleName:            {authtypes.Burner},
```
Comment (liamsi) [inline app/app.go:142]: Did you mean this by your comment?

**Comment by liamsi** [comment]
I think these are all the changes necessary to add back the governance module. While modifying the code, I've realized that we could restructure files/packages slightly to make it more readable/accessible. I think we can slowly move away from the way starport generates and structures everything. But that is orthogonal to this PR.

Either way, turning this into ready for review now.

**Review by adlerjohn — DISMISSED** [review]
utACK

File: app/app.go [hunk 3]
```diff
@@ -297,6 +311,15 @@ func New(
 		appCodec, keys[ibchost.StoreKey], app.GetSubspace(ibchost.ModuleName), app.StakingKeeper, app.UpgradeKeeper, scopedIBCKeeper,
 	)
 
+	// register the proposal types
+	govRouter := govtypes.NewRouter()
+	govRouter.AddRoute(govtypes.RouterKey, govtypes.ProposalHandler).
+		AddRoute(paramproposal.RouterKey, params.NewParamChangeProposalHandler(app.ParamsKeeper)).
+		// TODO: do we want a community pool at all? If not, does it even make sense to include that NewCommunityPoolSpendProposalHandler?
```
Comment (liamsi) [inline app/app.go:318]:
```suggestion
		// TODO: do we want a community pool at all? If not, does it even make sense to include that NewCommunityPoolSpendProposalHandler?
```

File: app/app.go (hunk 2 above)
Comment (evan-forbes) [reply app/app.go:142]: yep! everything should be good to go

**Review by evan-forbes — DISMISSED** [review]
:+1:

**Comment by evan-forbes** [comment]
fairly certain that the linter is only failing because of what was already fixed in https://github.com/celestiaorg/celestia-app/pull/273

**Comment by evan-forbes** [comment]
> While modifying the code, I've realized that we could restructure files/packages slightly to make it more readable/accessible.

I agree. I like what osmosis does here by separating the modules out into their own file. We could likely do this with even more components of the app.

https://github.com/osmosis-labs/osmosis/blob/main/app/modules.go

**Comment by liamsi** [comment]
> fairly certain that the linter is only failing because of what was already fixed in https://github.com/celestiaorg/celestia-app/pull/273

Let me rebase to be more certain :)

> I like what osmosis does here by separating the modules out into their own file. We could likely do this with even more components of the app.

Yes, exactly. I also thought Osmosis was a good example here!

**Comment by liamsi** [comment]
> fairly certain that the linter is only failing because of what was already fixed in https://github.com/celestiaorg/celestia-app/pull/273

confirmed, thanks

File: app/app.go (hunk 3 above)
Comment (liamsi) [inline app/app.go:318]: Actually. Let's remove this TODO. We should rather track this in an issue if anything. We had a community pool all along btw (with no governance to govern the tokens in that pool 😬).
```suggestion
		// TODO: do we want a community pool at all? If not, does it even make sense to include that NewCommunityPoolSpendProposalHandler?
```

**Review by adlerjohn — APPROVED** [review]
utACK

**Pull request merged**